### PR TITLE
Github workflow for nightly builds

### LINKED ISSUE #291: Create github workflow for nightly image builds
The goal of this issue is to create Github workflow which will build notebook image with SDK nightly. Notebook image will be published on Quay with the tag "latest". No other artifacts such as Python package or Github release will be created.

### PR DESCRIPTION
**Pull request by jiripetrlik**
# Issue link

Closes #291 
# What changes have been made

New workflow was created. It will run each day and produce nightly image for notebook.

# Verification steps
It was tested using my local fork: https://github.com/jiripetrlik/codeflare-sdk/actions/runs/5857794950/job/15880432547

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [*] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: .github/workflows/nightly-image-build.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+              run: poetry build
+            - name: Copy SDK package
+              run: cp dist/${SDK_FILE} custom-nb-image
+            - name: Modify Dockerfile to use locally build SDK
+              run: |
+                sed -i "s/ARG SDK_VERSION=.*/COPY ${{ env.SDK_FILE }} ./" custom-nb-image/Dockerfile
+                sed -i "s/codeflare-sdk==.*/${{ env.SDK_FILE }}\\\\/" custom-nb-image/Dockerfile
+            - name: Image Build
+              run: |
+                cd custom-nb-image
```
Comment (sutaakar) [inline .github/workflows/nightly-image-build.yaml:41]: You can use `working-directory` instead - see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun

File: .github/workflows/nightly-image-build.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+name: Nightly Image Build
+
+on:
+    workflow_dispatch:
+    schedule:
```
Comment (sutaakar) [inline .github/workflows/nightly-image-build.yaml:5]: Can you rather use `push` event (i.e. https://github.com/project-codeflare/codeflare-operator/blob/main/.github/workflows/operator-image.yml#L6)?
That way the `dev` image is refreshed immediately.

**Comment by jiripetrlik** [comment]
@sutaakar Thank you for your comments. I tried to address all of them.

@anishasthana @KPostOffice Hello, can you please take a look and merge if possible?

**Review by sutaakar — APPROVED** [review]

File: .github/workflows/nightly-image-build.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+              uses: abatilo/actions-poetry@v2
+              with:
+                poetry-version: ${{ env.POETRY_VERSION }}
+            - name: Run poetry install
+              run: poetry install --with docs
+            - name: Run poetry build
+              run: poetry build
+            - name: Copy SDK package
+              run: cp dist/${SDK_FILE} custom-nb-image
+            - name: Modify Dockerfile to use locally build SDK
```
Comment (Maxusmusti) [inline .github/workflows/nightly-image-build.yaml:36]: `build` -> `built`

**Review by Maxusmusti — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: .github/workflows/nightly-image-build.yaml (hunk 3 above)
Comment (Maxusmusti) [reply .github/workflows/nightly-image-build.yaml:36]: For line 36

File: .github/workflows/nightly-image-build.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+              run: |
+                sed -i "s/ARG SDK_VERSION=.*/COPY ${{ env.SDK_FILE }} ./" custom-nb-image/Dockerfile
+                sed -i "s/codeflare-sdk==.*/${{ env.SDK_FILE }}\\\\/" custom-nb-image/Dockerfile
+            - name: Image Build
+              working-directory: custom-nb-image
+              run: docker build -t quay.io/${{ env.QUAY_ORGANIZATION }}/notebook:${{ env.IMAGE_TAG }} .
```
Comment (anishasthana) [inline .github/workflows/nightly-image-build.yaml:42]: nit: I'd prefer to use podman here instead of docker, but I won't block the PR on it.

**Comment by jiripetrlik** [comment]
@Maxusmusti 
Thank you for the review. Typo should be fixed and PR was rebased.

**Review by Maxusmusti — APPROVED** [review]
LGTM

**Pull request merged**